src/lib/seam/components/AccessCodeForm/AccessCodeForm.tsx

```diff
@@ -0,0 +1,196 @@
+import classNames from 'classnames'
+import { useState } from 'react'
+
+import { getRandomInt } from 'lib/numbers.js'
+import { useCreateAccessCode } from 'lib/seam/access-codes/use-create-access-code.js'
+import { useDevice, type UseDeviceData } from 'lib/seam/devices/use-device.js'
+import { Button } from 'lib/ui/Button.js'
+import { FormField } from 'lib/ui/FormField.js'
+import { InputLabel } from 'lib/ui/InputLabel.js'
+import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
+import { TextField } from 'lib/ui/TextField/TextField.js'
+import { useToggle } from 'lib/ui/use-toggle.js'
+
+const minCodeLength = 4
+const maxCodeLength = 8
+
+export interface AccessCodeFormProps {
+  className?: string
+  onBack?: () => void
+  deviceId: string
+}
+
+export function AccessCodeForm({
+  className,
+  onBack,
+  deviceId,
+}: AccessCodeFormProps): JSX.Element | null {
+  const { device } = useDevice({
+    device_id: deviceId,
+  })
+
+  if (device == null) {
+    return null
+  }
+
+  return (
+    <div className={classNames('seam-access-code-form', className)}>
+      <Content className={className} onBack={onBack} device={device} />
+    </div>
+  )
+}
+
+function Content({
+  onBack,
+  device,
+}: Omit<AccessCodeFormProps, 'deviceId'> & {
+  device: NonNullable<UseDeviceData>
+}): JSX.Element {
+  const [name, setName] = useState('')
+  const [code, setCode] = useState('')
+  const [codeInputFocused, toggleCodeInputFocused] = useToggle()
+
+  const createAccessCode = useCreateAccessCode(device)
+
+  const save = (): void => {
+    if (name === '' || code === '') {
+      return
+    }
+
+    if (createAccessCode.isLoading) {
+      return
+    }
+
+    createAccessCode.mutate(
+      {
+        name,
+        code,
+      },
+      {
+        onSuccess: () => {
+          onBack?.()
+        },
+      }
+    )
+  }
+  const nameError = (): string | undefined => {
+    if (name.length > 60) {
+      return t.overCharacterLimitError
+    }
+    return undefined
+  }
+
+  const codeError = (): string | undefined => {
+    // Only check code on any input
+    if (code.trim().length === 0) {
+      return undefined
+    }
+
+    // If a non-digit exists
+    if (/\D/.test(code)) {
+      return 'Number only.'
+    }
+
+    if (code.length < minCodeLength || code.length > maxCodeLength) {
+      return 'Code must be between 4, and 8 digits.'
+    }
+
+    return undefined
+  }
+
+  const generateCode = (): void => {
```

Comment:
@dawnho Can you give some context on this feature? I have some concerns about generating a random code this way if we don't consider the device restrictions. It will generate code that fail to create and the user will be confused why the auto generation doesn't work. I recommend we remove this from the initial release and add it back after a separate discussion.